sdk/storage/storage-file-datalake/review/storage-file-datalake.api.md

```diff
@@ -295,6 +296,18 @@ export interface FileCreateOptions extends PathCreateOptions {
 export interface FileCreateResponse extends PathCreateResponse {
 }
 
+// @public
+export enum FileExpiryMode {
```

Comment:
could we use string literals for this instead? 

e.g.

```ts
type FileExpiryMode = "Absolute" | "NeverExpire" | "RelativeToCreation" | "RelativeToNow";
```

Reply:
It's from the generated code, I will see if I can get a transform to work this out.

Reply:
Changed swagger/README.md to do this. But it also changed some other enum interfaces.
Do we consider this as a breaking change? I can revert it if wanted.

Reply:
This is most definitely a breaking change - string unions don't have any runtime value associated with them, which is why we should prefer to use them, but if you've shipped enums, moving to string literal union is a breaking change. HOWEVER, you can keep the enums for people who want them, and change the signature from taking the enum to taking the string literal union. Best of both worlds, perhaps.

Reply:
This is a new type added in this PR so should be good?

Reply:
please discard my previous comments. I saw other changes now.

Reply:
@bterlson 
Are you suggesting we do this for old enums:

Take `PathResourceType` for example,

Keep the original enum.

```ts
export enum PathResourceType {
    // (undocumented)	
    Directory = "directory",	
    // (undocumented)	
    File = "file"	
}
```

Use the string union for function signature. Rename the union as the name conflicts.
```ts
import PathResourceType as PathResourceTypeModel from "generated/XXX";

export class DataLakeDirectoryClient extends DataLakePathClient {
    create(resourceType: PathResourceTypeModel, options?: PathCreateOptions): Promise<PathCreateResponse>;
...
}
```